imap/imapd.c

```diff
@@ -11829,7 +11829,6 @@ static int xfer_finalsync(struct xfer_header *xfer)
     if (!r && xfer->userid) {
         r = sync_do_user_seen(&xfer->sync_cs, xfer->userid, replica_seen);
         if (!r) r = sync_do_user_sub(&xfer->sync_cs, xfer->userid, replica_subs);
-        if (!r) r = sync_do_user_sieve(&xfer->sync_cs, xfer->userid, replica_sieve);
```

Comment:
I think we're able to find out the version of the destination, aren't we?  So maybe it could be conditional:  if the destination is new enough, don't replicate sieve, but if it's older, it still needs to be replicated.

Ahh, it's `backend_version()` that's used for figuring out the other end's version... and it's based on MAILBOX_MINOR_VERSION (currently 17)

Reply:
@elliefm @brong I reinstated the sievedir replication code and added a backend_supports_sieve_mailbox() check to NOT do it if the backend is v3.5+.  Cass still passes, but I haven't done any mixed version testing.  Please review.